common/lib/xmodule/xmodule/modulestore/tests/test_mongo_call_count.py

```diff
@@ -88,13 +88,21 @@ class CountMongoCallsCourseTraversal(TestCase):
     """
 
     @ddt.data(
-        (MIXED_OLD_MONGO_MODULESTORE_BUILDER, None, 189),  # The way this traversal *should* be done.
-        (MIXED_OLD_MONGO_MODULESTORE_BUILDER, 0, 387),     # The pathological case - do *not* query a course this way!
-        (MIXED_SPLIT_MODULESTORE_BUILDER, None, 7),  # The way this traversal *should* be done.
-        (MIXED_SPLIT_MODULESTORE_BUILDER, 0, 145)    # The pathological case - do *not* query a course this way!
+        # The way this traversal *should* be done (if you'll eventually load all the definitions anyway).
+        (MIXED_OLD_MONGO_MODULESTORE_BUILDER, None, False, 189),
+        (MIXED_OLD_MONGO_MODULESTORE_BUILDER, None, True, 189),
+        (MIXED_OLD_MONGO_MODULESTORE_BUILDER, 0, False, 387),
+        # The pathological case - do *not* traverse a course this way!
+        (MIXED_OLD_MONGO_MODULESTORE_BUILDER, 0, True, 387),
+        # The way this traversal *should* be done (if you'll eventually load all the definitions anyway).
+        (MIXED_SPLIT_MODULESTORE_BUILDER, None, False, 4),
+        (MIXED_SPLIT_MODULESTORE_BUILDER, None, True, 4),
+        # The pathological case - do *not* traverse a course this way!
+        (MIXED_SPLIT_MODULESTORE_BUILDER, 0, False, 143),
+        (MIXED_SPLIT_MODULESTORE_BUILDER, 0, True, 4)
```

Comment:
Consider adding `ddt` tests for `depth=2` since LMS uses this depth as an argument to `get_course` in multiple places.